src/js/wrapper.ts

```diff
@@ -109,7 +104,7 @@ export const NATIVE: SentryNativeWrapper = {
         bytesPayload = [...itemPayload];
       } else {
         bytesPayload = utf8ToBytes(JSON.stringify(itemPayload));
-        isFatalCrash = 'level' in itemPayload && itemPayload.level === 'fatal';
+        hardCrashed = (isEvent(itemPayload) && isHardCrash(itemPayload)) || hardCrashed;
```

Comment:
Why do we need the `isEvent` and the `|| hardCrashed`? I'm a bit confused.

Reply:
So I was thinking either:
- make `isHardCrash` accept all the payloads and internally for not Event payloads resolve `false`
- or make `isHardCrash` accept only Event because it's the only payload holding error info, but then I need to first check the type of the payload before passing it forward.

- one resolves on cleaner use at this line
- the other in cleaner function

Reply:
And the `|| hardCrashed` is because there can be multiple items in the envelope but only one is a  hard crash.

Reply:
See https://github.com/getsentry/sentry-react-native/pull/2480/files#r974163085